diff --git a/node_modules/@capgo/native-purchases/ios/Sources/NativePurchasesPlugin/NativePurchasesPlugin.swift b/node_modules/@capgo/native-purchases/ios/Sources/NativePurchasesPlugin/NativePurchasesPlugin.swift
index 7cb921e..71a2795 100644
--- a/node_modules/@capgo/native-purchases/ios/Sources/NativePurchasesPlugin/NativePurchasesPlugin.swift
+++ b/node_modules/@capgo/native-purchases/ios/Sources/NativePurchasesPlugin/NativePurchasesPlugin.swift
@@ -6,6 +6,7 @@ import StoreKit
  * Please read the Capacitor iOS Plugin Development Guide
  * here: https://capacitorjs.com/docs/plugins/ios
  */
+@MainActor
 @objc(NativePurchasesPlugin)
 public class NativePurchasesPlugin: CAPPlugin, CAPBridgedPlugin {
     public let identifier = "NativePurchasesPlugin"
@@ -23,7 +24,7 @@ public class NativePurchasesPlugin: CAPPlugin, CAPBridgedPlugin {
     ]
 
     private let pluginVersion: String = "7.15.3"
-    private var transactionUpdatesTask: Task<Void, Never>?
+    nonisolated(unsafe) private var transactionUpdatesTask: Task<Void, Never>?
 
     @objc func getPluginVersion(_ call: CAPPluginCall) {
         call.resolve(["version": self.pluginVersion])
